lib/formalist/elements/many_forms.rb

```diff
@@ -0,0 +1,113 @@
+require "formalist/element"
+require "formalist/child_forms/builder"
+
+module Formalist
+  class Elements
+    class ManyForms < Element
+      attribute :action_label
+      attribute :sortable
+      attribute :moveable
+      attribute :label
+      attribute :max_height
+      attribute :placeholder
+      attribute :embeddable_forms
+      attribute :validation
+      attribute :allow_create, default: true
+      attribute :allow_update, default: true
+      attribute :allow_destroy, default: true
+      attribute :allow_reorder, default: true
+
+      # FIXME: it would be tidier to have a reader method for each attribute
+      def attributes
+        super.merge(embeddable_forms: embeddable_forms_ast)
+      end
+
+       # @api private
+       def fill(input: {}, errors: {})
+        input = input[name] || []
+        errors = errors[name].to_a
+
+        children = child_form_builder.(input)
+
+        super(
+          input: input,
+          errors: errors,
+          children: children,
+        )
+      end
+
+      # Replace the form objects with their AST
+      def embeddable_forms_ast
+        @attributes[:embeddable_forms].to_h.map { |key, attrs|
+          [
+            key,
+            attrs.merge(
+              form: attrs[:form].to_ast,
+              attributes_template: Element::Attributes.new(attrs).to_ast
+            )
+          ]
+        }.to_h
+      end
+
+      def child_form_builder
+        ChildForms::Builder.new(@attributes[:embeddable_forms])
+      end
+
+      # Converts a collection of "many" repeating elements into an abstract
+      # syntax tree.
+      #
+      # It takes the following format:
+      #
+      # ```
+      # [:many, [params]]
```

Comment:
I _think_ there might be some copy/paste errors with this doc comment. Have highlighted them but feel free to let me know if I'm wrong!

```suggestion
      # [:many_forms, [params]]
```